external/fv3fit/fv3fit/reservoir/adapters.py

```diff
@@ -198,3 +206,97 @@ def load(cls, path: str) -> "HybridReservoirDatasetAdapter":
             model=model,
         )
         return adapter
+
+
+ReservoirAdapterType = Union[ReservoirDatasetAdapter, HybridReservoirDatasetAdapter]
+ReservoirModelLike = Union[ReservoirModelType, ReservoirAdapterType]
+
+
+@typing.no_type_check
+def split_multi_subdomain_model(
+    model: ReservoirModelLike,
+) -> Sequence[ReservoirModelLike]:
+    """ Split a multi-subdomain model into a list of single subdomain models.
+    """
+    is_adapter = isinstance(model, ReservoirDatasetAdapter) or isinstance(
+        model, HybridReservoirDatasetAdapter
+    )
+    if is_adapter:
+        adapter = model
+        model = adapter.model
+        is_adapter = True
+    else:
+        adapter = None
+        is_adapter = False
+
+    rank_divider = model.rank_divider
+    readout = model.readout
+
+    if rank_divider.n_subdomains == 1:
+        raise ValueError("Model must have multiple subdomains to split.")
+
+    new_rank_divider = RankXYDivider(
+        subdomain_layout=(1, 1),
+        overlap=rank_divider.overlap,
+        overlap_rank_extent=rank_divider.subdomain_extent,
+        z_feature_size=rank_divider._z_feature_size,
+    )
+
+    new_kwargs = {
+        "input_variables": model.input_variables,
+        "output_variables": model.output_variables,
+        "reservoir": model.reservoir,
+        "transformers": model.transformers,
+        "square_half_hidden_state": model.square_half_hidden_state,
+    }
+
+    if isinstance(model, HybridReservoirComputingModel):
+        new_kwargs["hybrid_variables"] = model.hybrid_variables
+
+    new_models = []
+    for i in range(rank_divider.n_subdomains):
```

Comment:
This doesn't have to happen in this PR if it's a pain, but  it may be useful to be able to split into smaller ranks that still contain >1 subdomain. e.g. original model covering a full tile contains 36 subdomains of 8x8 each and is split into 4 models, each with 9 subdomains.

Reply:
I think this is an interesting idea.  I think it can be accomplished using the rank divider on a subdomain layout indices to keep the coefficients grouped properly for the readout, but too complicated for this PR.